### ISSUE TITLE
Hide "import Xxxx from 'mystuff/lib/Xxxx.js'"

### ISSUE THREAD

**Issue by ibc** [esdoc/esdoc#441]
I don't want to publicly expose how to `import` a specific `class` in my library. Instead, my library exposes a factory.

So, how to avoid ESdoc displaying `import Xxxx from 'mystuff/lib/Xxxx.js'` at the top of the Xxxx `class` documentation?

**Comment by pzuraq** [comment]
You may need to modify the HTML generator for this, I don't think there is a public way to exclude it.

**Comment by ibc** [comment]
Yes, in fact I've found nothing to achieve that. However, I wonder: who assumes that **any** ES6 class within a library should be publicly exposed? In many cases, those are just for internal usage or, even if exposed via public API, the user is not supposed to directly `import` such a class, but instead get class instances vía other public APIs such as factories and so on.

**Comment by pzuraq** [comment]
Agreed. The plugin system in v1 should be powerful enough to handle this though, you can add arbitrary tags so if we can modify the HTML generation plugin we could pretty easily add it:

https://github.com/esdoc/esdoc-plugins/tree/master/esdoc-publish-html-plugin

I'm also in need of some custom functionality, would definitely be down to join forces and work on a fork of the publish plugin that's more flexible and feature-ful.

**Comment by ibc** [comment]
> https://github.com/esdoc/esdoc-plugins/tree/master/esdoc-publish-html-plugin

Most of the official plugins in https://github.com/esdoc/esdoc-plugins/tree/master/ don't even have a minimal description. I just don't know what they provide at all.

**Comment by pzuraq** [comment]
yeah, I had to dig in myself 😕  @h13i32maru seems to be backlogged, and is focussing most efforts on v1. I think the plugin system is a great decision in that regard, it'll make it so the community can add more functionality independently.

So, ESDoc itself will crawl the code and generate JSON which describes each thing. This JSON has an entry for each export, function, variable, class, etc. And each entry has some predefined tags, but also will have an array of arbitrary tags. So if I did:

```
/**
 * @private
 */
export function foo() { ... }
```

Or something like that, `private` would show up in that list.

The various plugins the in that repo then transform that JSON payload, some of them add docs for decorators and class properties such for instance. The publish-html plugin actually generates the docs from the final JSON description.

I [forked it here](https://github.com/pzuraq/esdoc-publish-module-html-plugin) to add the ability to have all the functions/const described in a single utility module be on their own page. Before, all the functions were on a single, package-wide page, as were the constants.

**Comment by ibc** [comment]
Thanks for he explanation.
